libs/libpam/patches/0007-Replace-strndupa-with-strndup.patch

```diff
@@ -0,0 +1,33 @@
+From 7030a7b9259d6e031d583af9cf55d7ad06987230 Mon Sep 17 00:00:00 2001
+From: Rosen Penev <rosenp@gmail.com>
+Date: Sun, 15 Jul 2018 20:43:44 -0700
+Subject: [PATCH] Replace strndupa with strndup
+
+glibc only.
+
+Signed-off-by: Rosen Penev <rosenp@gmail.com>
+---
+ modules/pam_exec/pam_exec.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/modules/pam_exec/pam_exec.c b/modules/pam_exec/pam_exec.c
+index 0ab6548..a003124 100644
+--- a/modules/pam_exec/pam_exec.c
++++ b/modules/pam_exec/pam_exec.c
+@@ -178,11 +178,11 @@ call_exec (const char *pam_type, pam_handle_t *pamh,
+ 		}
+ 
+ 	      pam_set_item (pamh, PAM_AUTHTOK, resp);
+-	      authtok = strndupa (resp, PAM_MAX_RESP_SIZE);
```

Comment:
Shouldn't in that case free the output?

Reply:
It may be easier to create a fixed buffer of pam_max_resp_size and copy there.

Reply:
@neheb @nmav 
Any update on this PR?

Reply:
Not sure how to resolve this. My C knowledge is limited. A fix should probably be upstreamed first.